modules/statistics/php/stats_demographic.class.inc

```diff
@@ -55,38 +56,40 @@ class Stats_Demographic extends \NDB_Form
     /**
      * RenderStatsTable function
      *
-     * @param string $sectionHeader     the value of sectionHeader
-     * @param string $tableHeader       the value of tableHeader
-     * @param string $subcats           the value of subcats
-     * @param string $visits            the value of visits
-     * @param string $dropdown_name     the value of dropdown_name
-     * @param string $dropdown_opt      the value of dropdown_opt
-     * @param string $dropdown_selected the value of dropdown_selected
-     * @param string $centres           the value of centres
-     * @param string $data              the value of data
-     * @param string $Subsection        the value of Subsection=''
-     * @param string $disclamer         the value of disclamer=''
-     * @param string $projectID         the value of projectID is null
+     * @param string $sectionHeader    the value of sectionHeader
+     * @param string $tableHeader      the value of tableHeader
+     * @param string $subcats          the value of subcats
+     * @param string $visits           the value of visits
+     * @param string $dropdownName     the value of dropdownName
+     * @param string $dropdownOpt      the value of dropdownOpt
+     * @param string $dropdownSelected the value of dropdownSelected
+     * @param string $centers          the value of centers
+     * @param string $data             the value of data
+     * @param string $subsection       the value of subsection=''
+     * @param string $disclamer        the value of disclamer=''
+     * @param string $projectID        the value of projectID is null
      *
      * @return string
      */
     function renderStatsTable($sectionHeader,
         $tableHeader,
         $subcats,
         $visits,
-        $dropdown_name,
-        $dropdown_opt,
-        $dropdown_selected,
-        $centres,
+        $dropdownName,
+        $dropdownOpt,
+        $dropdownSelected,
+        $centers,
         $data,
-        $Subsection="",
+        $subsection="",
         $disclamer='',
         $projectID=null
     ) {
-
-        $tpl_data = array();
+        // This line replaces the empty $projectID string with null
+        // `$projectID ?? null` would retain empty strings.
```

Comment:
i don't think this line is necessary. line 87 is good enough of an explanation of what you want to do, i think

Reply:
@zaliqarosli This comment also applies to your comment below.

Reply:
@johnsaigle i think you misunderstanding me. I understand what the line does - it is explained with line 87. i just don't think the addition line 88 is needed. i think what @ZainVirani did here with the comments is more than enough

Reply:
I think its perfect.